Bonnie Lake Competition Controller changes based on input from John & Asher

Samsuppp opened the pull request.

mmcadamsms committed:
Updates to core operations that I think will be needed to meet Josies ask of having shooter speed track shooter angle.

Samsuppp committed:
Update Manual Input for Bonny Lake

Updated based on Driver input to new controller configuration

mmcadamsms commented on ted/src/main/java/frc/robot/control/ManualInputInterfaces.java:
```diff
@@ -192,92 +189,117 @@ private void bindCommandsToDriverXboxButtons(){
                 "Shoot at speed!!")
```
Simeon,

If we were to use the existing ShooterShootCommand, it uses a static default speed for shooter outfeed.  This would prove itself to not be correct outfeed speeds for all shot angles.  I think the block above should actually change so that the double supplier constructors can be used instead based on the ShooterOutfeedSpeedProvider thing I wrote (in the /common directory).  So lines 184-186 should be replaced with something more like:

              new ShooterShootCommand(
                () -> this.outfeedSpeedProvider.getShotSpeedForCurrentAngle(),
                () -> this.outfeedSpeedProvider.getShotSpeedForCurrentAngle(),
                subsystemCollection.getShooterOutfeedSubsystem(), 
                subsystemCollection.getFeederSubsystem()),

What this will do is allow the speeds to change based on the angle that the shooter actually is at when the shot happens.

Thanks,
Mike

Samsuppp replied:
Code added

mmcadamsms reviewed (commented):
**[UPDATE]** I think next all you need to do is test that the new controls work on the robot now.  Maybe if you get time after limelight testing tonight you and Asher can test the commands with controllers on the actual robot?

I made one comment. **[UPDATE]** thanks for the update

We may also want to eventually have Asher update his autos to use the updated constants.  But I guess I can add that here now too. **[UPDATE]** I updated Asher's auto code to use the constants instead of the magic numbers.

mmcadamsms committed:
Update auto chooser to switch to using constants instead of using magic numbers.

Samsuppp committed:
Update ManualInputInterfaces.java

mcadamswa committed:
Update Constants.java

mmcadamsms committed:
Attempt fix for shooter angle pov up/down

Based on conversation with Simeon last night it appears his change on POV didn't accomplish arm angle up down.  I think what might be going on is that the angle went up very little (or down very little) and then the angle was within tolerance and the ShooterSetAngle command was done.  Because .whileTrue does 'one and done' functionality the angle change stopped (I wasn't aware that .whileTrue only runs a command until its completion ... I had thought it would rerun the same command).  Change here adds a RepeatCommand to continue attempting rerun of the ShooterSetAngleTesterCommand in series while the button is still pressed.

mmcadamsms reviewed (approved):
tested today with modified robot ... works as expected now
